File: plugins/go/clock.5s.go
// +build ignore

package main

import (
	"fmt"
	"time"
)

func main() {
	now := time.Now()
	hour := now.Hour()

	// Determine icon based on time of day
	var icon string
	switch {
	case hour >= 6 && hour < 12:
		icon = "\U0001F305" // sunrise
	case hour >= 12 && hour < 18:
		icon = "\u2600\uFE0F" // sun
	case hour >= 18 && hour < 21:
		icon = "\U0001F307" // sunset
	default:
		icon = "\U0001F319" // moon
	}

	// Format time
	timeStr := now.Format("15:04:05")

	fmt.Printf("%s %s\n", icon, timeStr)
	fmt.Println("---")
	fmt.Printf("Date: %s\n", now.Format("Monday, January 2, 2006"))
	_, week := now.ISOWeek()
	fmt.Printf("Week: %d\n", week)
	fmt.Printf("Day of Year: %d\n", now.YearDay())
	fmt.Println("---")
	fmt.Println("Refresh | refresh=true")
}

File: plugins/clock.5s.rs
use std::process::Command;
use std::str;
use std::str::FromStr;

fn main() {
    // Use Crossbar CLI API to get current time
    let output_time = Command::new("crossbar")
        .arg("time")
        .arg("24h")
        .output();

    let time_str = match output_time {
        Ok(cmd_output) => str::from_utf8(&cmd_output.stdout).unwrap_or("N/A").trim().to_string(),
        Err(_) => "N/A".to_string(),
    };

    // Use Crossbar CLI API to get current date
    let output_date = Command::new("crossbar")
        .arg("date")
        .output();

    let date_str = match output_date {
        Ok(cmd_output) => str::from_utf8(&cmd_output.stdout).unwrap_or("N/A").trim().to_string(),
        Err(_) => "N/A".to_string(),
    };

    // Try to parse hour for icon logic
    let mut hour: i32 = -1;
    if let Some(h_str) = time_str.split(':').next() {
        if let Ok(h) = i32::from_str(h_str) {
            hour = h;
        }
    }

    // Determine icon based on time of day
    let icon = match hour {
        6..=11 => "\u{1F305}", // sunrise
        12..=17 => "\u{2600}\u{FE0F}", // sun
        18..=20 => "\u{1F307}", // sunset
        _ => "\u{1F319}", // moon
    };

    println!("{} {}", icon, time_str);
    println!("---");
    println!("Date: {}", date_str);
    // Removed Week and Day of Year as crossbar CLI doesn't provide these directly in a simple format
    println!("---");
    println!("Refresh | refresh=true");
}
